src/Core/Services/StateService.cs

```diff
@@ -1001,17 +1001,25 @@ public async Task SetPushInitialPromptShownAsync(bool? value)
             await SetValueAsync(key, value, options);
         }
 
-        public async Task<DateTime?> GetPushLastRegistrationDateAsync()
+        public async Task<DateTime?> GetPushLastRegistrationDateAsync(string userId = null)
         {
+            if (userId == null)
```

Comment:
These getters should follow the established pattern from other account-specific getters like so:

`var reconciledOptions = ReconcileOptions(new StorageOptions { UserId = userId }, await GetDefaultStorageOptionsAsync());`
`var key = Constants.PushLastRegistrationDateKey(reconciledOptions.UserId);`
`return await GetValueAsync<DateTime?>(key, reconciledOptions);`

Reply:
@mp-bw, the other examples for account-specific getters don't set the `userId` to be the active user if it comes in as `null`, right?  This was a recommendation that @fedemkr had, to take the lookup for active user out of the calling code.  I originally had the same pattern, but it required doing the check for active user outside of each call to these methods.

I'm fine either way, I just want to make sure it fits with best practices.

Reply:
By design all calls to StateService methods automatically use the active userId if a `null` userId is sent from the calling code (this happens in `ReconcileOptions` via the object returned from `GetDefaultStorageOptionsAsync`).  I think Fede's suggestion was to follow that pattern, but I'm not sure if I'm misreading something.  @fedemkr ?

Reply:
> I'd keep the userId = null parameter on the methods though.

Yes, that is required for this design to work.

Reply:
Thank you both, and apologies for misunderstanding the recommendation @fedemkr .  I've corrected this, and it is much cleaner now.